### PR TITLE
add insert token to support  rules based on streams

### PR DESCRIPTION
**Pull request by AlvaroVega**
https://github.com/telefonicaid/perseo-fe/issues/630

### CODE REVIEW

File: lib/myutils.js [hunk 1]
```diff
@@ -318,8 +319,8 @@ function removeCRLFchars(text) {
 
 function ruleWithContext(rule) {
     const ruleText = removeCRLFchars(rule.text);
-    const preSelectMatch = ruleText.match(/.+?(?=\bselect )/i);
-    const SelectMatch = ruleText.match(/\bselect .*/i);
+    const preSelectMatch = ruleText.match(/.+?(?=\bselect |insert )/i);
+    const SelectMatch = ruleText.match(/\bselect |insert .*/i);
```
Comment (mrutid) [inline lib/myutils.js:323]:
```suggestion
    const SelectMatch = ruleText.match(/\bselect .*|insert .*/i);
```

**Review by mrutid — CHANGES_REQUESTED** [review]

**Review by mrutid — APPROVED** [review]
LGTM

**Review by fgalan — APPROVED** [review]
LGTM

**Comment by AlvaroVega** [comment]
Trying to test a stream rule (just inserting 'insert into mystream' into an existing rule) with perseo-fe (latest) and perseo-core (latest):

my_no_action_rule
action: []
text: insert into mystream select "my_no_action_rule" as ruleName, ev.id? as id, ev.temperature__type? as type, ev.temperature? as value, * from pattern [every ev=iotEvent]


perseo-core logs:

time=2022-03-01T12:35:34.124Z | lvl=DEBUG | from=::ffff:172.17.0.15 | corr=2a7a87ed-1d85-4ece-a820-bed4e5c60bec; node=lF8smSAjD; perseocep=9 | trans=159e34b5-d365-4dcd-bc85-8bbc06914d2f | srv=smartcity | subsrv=/ | op=parse | comp=perseo-core | msg=.parse Parsing expr=insert into mynewstream  context ctxt$smartcity$ insert into mynewstream select "my_no_action_rule" as ruleName, ev.id? as id, ev.temperature__type? as type, ev.temperature? as value, * from pattern [every ev=iotEvent]
time=2022-03-01T12:35:34.133Z | lvl=DEBUG | from=::ffff:172.17.0.15 | corr=2a7a87ed-1d85-4ece-a820-bed4e5c60bec; node=lF8smSAjD; perseocep=9 | trans=159e34b5-d365-4dcd-bc85-8bbc06914d2f | srv=smartcity | subsrv=/ | op=parse | comp=perseo-core | msg=Error parsing statement [insert into mynewstream  context ctxt$smartcity$ insert into mynewstream select "my_no_action_rule" as ruleName, ev.id? as id, ev.temperature__type? as type, ev.temperature? as value, * from pattern [every ev=iotEvent]]
org.antlr.v4.runtime.misc.ParseCancellationException: null
	at org.antlr.v4.runtime.BailErrorStrategy.recover(BailErrorStrategy.java:51) ~[antlr4-runtime-4.7.2.jar:4.7.2]
	at com.espertech.esper.compiler.internal.generated.EsperEPL2GrammarParser.startEPLExpressionRule(EsperEPL2GrammarParser.java:668) ~[esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.util.CompilerHelperSingleEPL$1.invokeParseRule(CompilerHelperSingleEPL.java:36) ~[esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.parse.ParseHelper.parse(ParseHelper.java:94) [esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.util.CompilerHelperSingleEPL.parseWalk(CompilerHelperSingleEPL.java:64) [esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.util.CompilerHelperSingleEPL.parseWalk(CompilerHelperSingleEPL.java:47) [esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.util.CompilerHelperStatementProvider.compileItem(CompilerHelperStatementProvider.java:92) [esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.util.CompilerHelperModuleProvider.compileToBytes(CompilerHelperModuleProvider.java:109) [esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.util.CompilerHelperModuleProvider.compile(CompilerHelperModuleProvider.java:82) [esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.util.EPCompilerImpl.compile(EPCompilerImpl.java:89) [esper-compiler-8.4.0.jar:?]
	at com.telefonica.iot.perseo.Utils.compileDeploy(Utils.java:412) [classes/:?]
	at com.telefonica.iot.perseo.RulesManager.make(RulesManager.java:138) [classes/:?]
	at com.telefonica.iot.perseo.RulesServlet.doPost(RulesServlet.java:125) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:681) [servlet-api.jar:4.0.FR]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764) [servlet-api.jar:4.0.FR]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227) [catalina.jar:9.0.58]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [catalina.jar:9.0.58]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) [tomcat-websocket.jar:9.0.58]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) [catalina.jar:9.0.58]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [catalina.jar:9.0.58]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197) [catalina.jar:9.0.58]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) [catalina.jar:9.0.58]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:540) [catalina.jar:9.0.58]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) [catalina.jar:9.0.58]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) [catalina.jar:9.0.58]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687) [catalina.jar:9.0.58]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) [catalina.jar:9.0.58]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:359) [catalina.jar:9.0.58]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399) [tomcat-coyote.jar:9.0.58]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:9.0.58]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:889) [tomcat-coyote.jar:9.0.58]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1735) [tomcat-coyote.jar:9.0.58]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:9.0.58]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-util.jar:9.0.58]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-util.jar:9.0.58]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:9.0.58]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: org.antlr.v4.runtime.NoViableAltException
	at org.antlr.v4.runtime.atn.ParserATNSimulator.noViableAlt(ParserATNSimulator.java:2026) ~[antlr4-runtime-4.7.2.jar:4.7.2]
	at org.antlr.v4.runtime.atn.ParserATNSimulator.execATN(ParserATNSimulator.java:467) ~[antlr4-runtime-4.7.2.jar:4.7.2]
	at org.antlr.v4.runtime.atn.ParserATNSimulator.adaptivePredict(ParserATNSimulator.java:393) ~[antlr4-runtime-4.7.2.jar:4.7.2]
	at com.espertech.esper.compiler.internal.generated.EsperEPL2GrammarParser.eplExpression(EsperEPL2GrammarParser.java:1659) ~[esper-compiler-8.4.0.jar:?]
	at com.espertech.esper.compiler.internal.generated.EsperEPL2GrammarParser.startEPLExpressionRule(EsperEPL2GrammarParser.java:660) ~[esper-compiler-8.4.0.jar:?]
	... 35 more
time=2022-03-01T12:35:34.181Z | lvl=ERROR | from=::ffff:172.17.0.15 | corr=2a7a87ed-1d85-4ece-a820-bed4e5c60bec; node=lF8smSAjD; perseocep=9 | trans=159e34b5-d365-4dcd-bc85-8bbc06914d2f | srv=smartcity | subsrv=/ | op=make | comp=perseo-core | msg=creating statement com.espertech.esper.common.client.EPException: Incorrect syntax near 'context' (a reserved keyword) at line 1 column 25 [insert into mynewstream  context ctxt$smartcity$ insert into mynewstream select "my_no_action_rule" as ruleName, ev.id? as id, ev.temperature__type? as type, ev.temperature? as value, * from pattern [every ev=iotEvent]]

**Pull request merged**